templates.go

```diff
@@ -0,0 +1,368 @@
+package resend
+
+import (
+	"context"
+	"net/http"
+)
+
+// VariableType represents the type of a template variable
+type VariableType string
+
+const (
+	VariableTypeString  VariableType = "string"
+	VariableTypeNumber  VariableType = "number"
+	VariableTypeBoolean VariableType = "boolean"
+	VariableTypeObject  VariableType = "object"
+	VariableTypeList    VariableType = "list"
+)
```

Comment:
We've decided to not add `boolean`, `object`, and `list` variable types yet (I noticed that the docs still didn't reflect this decision, sorry about the confusion)

I see some tests and the example might need to be updated as well to reflect that only `string` and `number` will be covered

```suggestion
const (
	VariableTypeString  VariableType = "string"
	VariableTypeNumber  VariableType = "number"
)
```